environments/environments.go

```diff
@@ -0,0 +1,142 @@
+package environments
+
+import (
+	"errors"
+	"sort"
+	"sync"
+
+	"github.com/airware/vili/kube"
+	"github.com/airware/vili/kube/v1"
+)
+
+var environments []Environment
+var rwMutex sync.RWMutex
+
+// Environment describes an environment backed by a kubernetes namespace
+type Environment struct {
+	Name      string `json:"name"`
+	Protected bool   `json:"protected"`
+	Prod      bool   `json:"prod"`
+	Approval  bool   `json:"approval"`
+}
+
+// Init initializes the global environments list
+func Init(envs []Environment) {
+	rwMutex.Lock()
+	environments = []Environment{}
+	for _, env := range envs {
+		environments = append(environments, env)
+	}
+	sort.Sort(byName(environments))
+	rwMutex.Unlock()
+}
+
+// Environments returns a snapshot of all of the known environments
+func Environments() (ret []Environment) {
+	rwMutex.RLock()
+	for _, env := range environments {
+		ret = append(ret, env)
+	}
+	rwMutex.RUnlock()
+	return
+}
+
+// Create creates a new environment with `name`
+func Create(name string) error {
+	_, status, err := kube.Namespaces.Create(&v1.Namespace{
```

Comment:
Yeah it won't accept invalid or duplicate namespaces, returning an `*unversioned.Status`